### PR TITLE
Disable SyncLoop, RetrieveLoop, and BlockStoreRetrieveLoop in Aggregator

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
## Overview
Closes: #1358 

## Checklist

- [x] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

## Summary by CodeRabbit

- **Refactor**
	- Enhanced node startup process to conditionally initiate different sets of operations for improved performance and reliability.
	- Streamlined task initiation by adding new routines and removing unnecessary return statements for a smoother startup sequence.

### CODE REVIEW

File: node/full.go [hunk 1]
```diff
@@ -341,10 +341,12 @@ func (n *FullNode) OnStart() error {
...
 		go n.blockPublishLoop(n.ctx)
+	} else {
+		go n.blockManager.RetrieveLoop(n.ctx)
+		go n.blockManager.BlockStoreRetrieveLoop(n.ctx)
+		go n.blockManager.SyncLoop(n.ctx, n.cancel)
 	}
-	go n.blockManager.RetrieveLoop(n.ctx)
-	go n.blockManager.BlockStoreRetrieveLoop(n.ctx)
-	go n.blockManager.SyncLoop(n.ctx, n.cancel)
+
```
Comment (tuxcanfly) [inline node/full.go:349]: Returning from the if condition reads slightly better:

```go
	if n.nodeConfig.Aggregator {
		n.Logger.Info("working in aggregator mode", "block time", n.nodeConfig.BlockTime)
		go n.blockManager.AggregationLoop(n.ctx, n.nodeConfig.LazyAggregator)
		go n.blockManager.BlockSubmissionLoop(n.ctx)
		go n.headerPublishLoop(n.ctx)
		go n.blockPublishLoop(n.ctx)
		return nil
	}

	go n.blockManager.RetrieveLoop(n.ctx)
	go n.blockManager.BlockStoreRetrieveLoop(n.ctx)
	go n.blockManager.SyncLoop(n.ctx, n.cancel)

	return nil
```

**Review by tuxcanfly — DISMISSED** [review]

File: node/full.go (hunk 1 above)
Comment (Manav-Aggarwal) [reply node/full.go:349]: true

**Review by tuxcanfly — APPROVED** [review]

File: node/full.go [hunk 2]
```diff
@@ -341,6 +341,7 @@ func (n *FullNode) OnStart() error {
 		go n.blockManager.BlockSubmissionLoop(n.ctx)
 		go n.headerPublishLoop(n.ctx)
 		go n.blockPublishLoop(n.ctx)
+		return nil
```
Comment (tzdybal) [inline node/full.go:344]: This is neat. I would say just a little bit _too neat_ ;)
Consider adding a short comment to explicitly state that **single** aggregator doesn't need `RetrieveLoop`, `BlockStoreRetrieveLoop` and `SyncLoop`.

**Review by tzdybal — APPROVED** [review]
LGTM

File: node/full.go (hunk 2 above)
Comment (Manav-Aggarwal) [reply node/full.go:344]: Added in the spec :)

**Pull request merged**